src/org/loklak/client/APIGenerator.java

```diff
@@ -0,0 +1,84 @@
+package org.loklak.client;
+
+import org.json.JSONObject;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.InvocationHandler;
+import java.lang.reflect.Method;
+import java.lang.reflect.Parameter;
+import java.lang.reflect.Proxy;
+
+import static org.loklak.tools.JsonIO.loadJson;
+import static org.loklak.tools.JsonIO.pushJson;
+
+public class APIGenerator {
+
+    // to restrict instantiation of class
+    private APIGenerator() {
+    }
+
+    public static <T> T createApiMethods(Class<T> service, final String baseUrl) {
+        return (T) Proxy.newProxyInstance(service.getClassLoader(), new Class<?>[]{service},
+                new InvocationHandler() {
+                    @Override
+                    public Object invoke(Object o, Method method, Object[] values)
+                            throws Throwable {
+                        Parameter[] params = method.getParameters();
+                        Object[] paramValues = values;
+
+                        /*
+                        format of annotation name:
+                        @packageWhereAnnotationIsDeclared$AnnotationName()
+                        Example: @org.loklak.client.LoklakAPI$GET()
+                        */
+                        Annotation annotation = method.getAnnotations()[0];
+                        String annotationName = annotation.toString().toLowerCase();
+
+                        String apiUrl = createGetRequestUrl(baseUrl, method.getName(),
+                                params, paramValues);
+                        if (annotationName.contains("get")) { // GET REQUEST
+                            return loadJson(apiUrl);
+                        } else { // POST REQUEST
+                            JSONObject jsonObjectToPush = (JSONObject) paramValues[0];
+                            String postRequestUrl = createPostRequestUrl(baseUrl, method.getName());
+                            return pushJson(postRequestUrl, params[0].getName(), jsonObjectToPush);
+                        }
+                    }
+                });
+    }
+
+    private static String createGetRequestUrl(
+            String baseUrl, String methodName, Parameter[] params, Object[] paramValues) {
```

Comment:
The method declaration line was crossing 100 chars. So, defined the params in a new line.